copy/copy.go

```diff
@@ -69,20 +69,24 @@ func (d *digestingReader) Read(p []byte) (int, error) {
 }
 
 // copier allows us to keep track of diffID values for blobs, and other
-// data, that we're copying between images, and cache other information that
-// might allow us to take some shortcuts
+// data shared across one or more images in a possible manifest list.
 type copier struct {
-	copiedBlobs       map[digest.Digest]digest.Digest
-	cachedDiffIDs     map[digest.Digest]digest.Digest
+	copiedBlobs      map[digest.Digest]digest.Digest
+	cachedDiffIDs    map[digest.Digest]digest.Digest
+	dest             types.ImageDestination
+	rawSource        types.ImageSource
+	reportWriter     io.Writer
+	progressInterval time.Duration
+	progress         chan types.ProgressProperties
+}
+
+// imageCopier tracks state specific to a single image (possibly an item of a manifest list)
+type imageCopier struct {
+	c                 *copier
```

Comment:
Should the object be a copier rather then a "c"

Reply:
It is used all over the place, and consistent with `func (c *copier)` receivers of the per-manifest-list code.